src/Paprika/Data/NibblePath.cs

```diff
@@ -209,9 +209,9 @@ private NibblePath(ref byte span, byte odd, byte length)
     /// </summary>
     /// <param name="destination">The destination to write to.</param>
     /// <returns>The leftover that other writers can write to.</returns>
-    public Span<byte> WriteToWithLeftover(Span<byte> destination)
+    public Span<byte> WriteToWithLeftover(Span<byte> destination, bool includeLengthAndOddity = true)
```

Comment:
This could be something that is a hot spot. I'd go with a specialized method like `WriteToWithLeftoverNoPreamble` or similar. Maybe you could come up with a less mouthful name 😅 Effectively, it'd be nice if it was branchless even if it requires a bit of duplication. You can take inspiration from `Node.Leaf` and the way it omits the preamble of the `NibblePath`.

Reply:
I'll split it back to two functions, with a small yet descriptive name ;)
And take a look at Node.Leaf for inspiration :)

Reply:
Took some time, but it should be working now :)
Although I have doubts about this particular block, I want to ask you tomorrow :
```
// clearing the oldest nibble, if needed
        // yes, it can be branch free
        if (((odd + length) & 1) == 0)
            return spanLength + PreambleLength;

        ref var oldest = ref destination[spanLength];
        oldest = (byte)(oldest & 0b1111_0000);

        return spanLength + PreambleLength;
```